idaes/core/surrogate/onnx_surrogate.py

```diff
@@ -0,0 +1,250 @@
+        representing the surrogate and to evaluate the surrogate a set of user
+        provided points.
+
+        This constructor should only be used when first creating the surrogate within IDAES.
+        Once created, this object can be stored to disk using save_to_folder and loaded
+        with load_from_folder
+
+        Args:
+           onnx_model: Onnx model file to be loaded.
+           input_labels: list of str
+              The ordered list of labels corresponding to the inputs in the keras model
+           output_labels: list of str
+              The ordered list of labels corresponding to the outputs in the keras model
+           input_bounds: None of dict of tuples
+              Keys correspond to each of the input labels and values are the tuples of
+              bounds (lb, ub)
+           input_scaler: None or OffsetScaler
+              The scaler to be used for the inputs. If None, then no scaler is used
+           output_scaler: None of OffsetScaler
+              The scaler to be used for the outputs. If None, then no scaler is used
+        """
+        super().__init__(
+            input_labels=input_labels,
+            output_labels=output_labels,
+            input_bounds=input_bounds,
+            input_scaler=input_scaler,
+            output_scaler=output_scaler,
+        )
+
+        self._onnx_model = onnx_model
+
+    class Formulation(Enum):
+        FULL_SPACE = 1
+        REDUCED_SPACE = 2
+        RELU_BIGM = 3
+        RELU_COMPLEMENTARITY = 4
+
+    def populate_block(self, block, additional_options=None):
+        """
+        Method to populate a Pyomo Block with the keras model constraints.
+
+        Args:
+           block: Pyomo Block component
+              The block to be populated with variables and/or constraints.
+           additional_options: dict or None
+              If not None, then should be a dict with the following keys;
+              'formulation': KerasSurrogate.Formulation
+              The formulation to use with OMLT. Possible values are FULL_SPACE,
+              REDUCED_SPACE, RELU_BIGM, or RELU_COMPLEMENTARITY (default is FULL_SPACE)
+        """
+        formulation = additional_options.pop(
+            "formulation", ONNXSurrogate.Formulation.REDUCED_SPACE
+        )
+        omlt_scaling, scaled_input_bounds = self.generate_omlt_scaling_objecets()
+
+        # omlt takes *scaled* input bounds as a dictionary with int keys
+        input_bounds = dict(enumerate(self.input_bounds().values()))
+        scaled_input_bounds = omlt_scaling.get_scaled_input_expressions(input_bounds)
+        scaled_input_bounds = {i: tuple(bnd) for i, bnd in scaled_input_bounds.items()}
+
+        # TODO: remove this once new OMLT 1.2 is made available and includes tanh support
+        # overrides default available activation functions for ONNX, tanh is not listed in 1.1 but is supported
+        omltio.onnx_parser._ACTIVATION_OP_TYPES = [
```

Comment:
Does this need catches for anything?

Reply:
Not sure best catches to implement here, thoughts on just checking for OMLT version? (It should not be needed with future OMLT releases.